dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt

```diff
@@ -0,0 +1,144 @@
+    private var v2MessageLogged: Boolean = false
+
+    fun logV1Warning() {
+        if (!v1WarningLogged) {
+            logger.warn(
+                """
+                |⚠ Warning: Dokka Gradle Plugin V1 mode is enabled
+                |
+                |  V1 mode is deprecated, and will be removed in Dokka version 2.0.20.
+                |
+                |  Please migrate Dokka Gradle Plugin to V2. This will require updating your project.
+                |  To get started check out the Dokka Gradle Plugin Migration guide
+                |      https://kotl.in/dokka-gradle-migration
+                |
+                |  Once you have prepared your project, enable V2 by adding
+                |      $V2_PLUGIN_ENABLED_FLAG=true
+                |  to your project's `gradle.properties`
+                |
+                |  Please report any feedback or problems to Dokka GitHub Issues
+                |      https://github.com/Kotlin/dokka/issues/
+                """.trimMargin().surroundWithBorder()
+            )
+            v1WarningLogged = true
+        }
+    }
+
+    fun logV2Message() {
+        if (!suppressV2PluginMessage && !v2MessageLogged) {
+            logger.lifecycle(
+                """
+                |Dokka Gradle Plugin V2 is enabled ♡
+                |
+                |  We would appreciate your feedback!
+                |  Please report any feedback or problems to Dokka GitHub Issues
+                |      https://github.com/Kotlin/dokka/issues/
+                |
+                |  If you need help or advice, check out the migration guide
+                |      https://kotl.in/dokka-gradle-migration
+                |
+                |  You can suppress this message by adding
+                |      $V2_PLUGIN_ENABLED_QUIET_FLAG=true
+                |  to your project's `gradle.properties`
+                """.trimMargin().surroundWithBorder()
+            )
+            v2MessageLogged = true
+        }
+    }
+
+    companion object {
+        private val logger = Logging.getLogger(PluginFeaturesService::class.java)
+
+        /**
+         * Register a new [PluginFeaturesService], or get an existing instance.
+         */
+        internal val Project.pluginFeaturesService: PluginFeaturesService
+            get() = gradle.sharedServices
+                .registerIfAbsent("PluginFeaturesService.internal", PluginFeaturesService::class) {
+                    parameters {
+                        enableV2Plugin.set(getFlag(V2_PLUGIN_ENABLED_FLAG))
+                        suppressV2PluginMessage.set(getFlag(V2_PLUGIN_ENABLED_QUIET_FLAG))
+                    }
+                }.get()
+
+        internal const val V2_PLUGIN_ENABLED_FLAG =
+            "org.jetbrains.dokka.experimental.gradlePlugin.enableV2"
+
+        internal const val V2_PLUGIN_ENABLED_QUIET_FLAG =
+            "$V2_PLUGIN_ENABLED_FLAG.quiet"
```

Comment:
minor: I do like suffix approach!
But, may be it's better to use `nowarn` suffix as in KGP flags, like `kotlin.wasm.stability.nowarn` or `kotlin.mpp.androidSourceSetLayoutV2AndroidStyleDirs.nowarn=true`?

Reply:
Yeah, I did think about that, but I'm on the fence because some of the messages are logged at LIFECYCLE level, not WARN. 'quiet' is more generic, and fits both. But `nowarn` would be consistent?

Hmmm I'll have a think.

Reply:
Done

Reply:
I added a `.noWarn` variant because I really really really don't want to be nagged about spelling.